### PR TITLE
139550-list-total-processed-payment-requests

### PR DESCRIPTION
**Pull request by simonnineteen90**
https://dev.azure.com/defragovuk/DEFRA-FFC/_sprints/taskboard/DEFRA-FFC%20Payments%20Team/DEFRA-FFC/Payment%20Service/Sprint%2006?workitem=139550

### CODE REVIEW

File: app/payments/get-payments-by-scheme-id.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const { SCHEME_ID } = require('../constants/categories')
+const { SCHEME_ID_VALUE } = require('../constants/scheme-id-value')
+const { getData } = require('./get-data')
+
+const getPaymentsBySchemeId = async (schemeId = SCHEME_ID_VALUE) => {
```
Comment (simonnineteen90) [inline app/payments/get-payments-by-scheme-id.js:5]: `getData` function requires a value property but not required anywhere else to get the events by scheme as filtering by category. 
Therefore used a default param here but unsure if there is a better way so open to suggestions here.

Comment (simonnineteen90) [reply app/payments/get-payments-by-scheme-id.js:5]: This could be adapted for future tickets to only return events for specific scheme however as the default view is to display all scheme data its unlikly this will be required and filtering to specific scheme will be done on frontend in my opinion.

File: app/routes/view-processed-payment-requests.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const config = require('../config')
+const { schemeAdmin, holdAdmin } = require('../auth/permissions')
+const { getPaymentsBySchemeId } = require('../payments')
+
+module.exports = [{
+  method: 'GET',
+  path: '/monitoring/view-processed-payment-requests',
+  options: {
+    auth: { scope: [schemeAdmin, holdAdmin] }
```
Comment (johnwatson484) [inline app/routes/view-processed-payment-requests.js:9]: Should have `dataView` role too

File: app/payments/get-payments-by-scheme-id.js (hunk 1 above)
Comment (Kirsten1822) [reply app/payments/get-payments-by-scheme-id.js:5]: So the schemeIdValue will always be 1? I'm not too sure I understand what this is doing

File: app/payments/get-payments-by-scheme-id.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const { SCHEME_ID } = require('../constants/categories')
+const { SCHEME_ID_VALUE } = require('../constants/scheme-id-value')
```
Comment (Kirsten1822) [inline app/payments/get-payments-by-scheme-id.js:2]: I'd also swap these two around just to match the order they appear below

File: test/integration/narrow/routes/monitoring.test.js [hunk 4]
```diff
@@ -261,4 +262,61 @@ describe('monitoring test', () => {
...
+      method: 'GET',
+      url: '/monitoring/view-processed-payment-requests',
+      auth
+    }
+
+    const response = await server.inject(options)
+    expect(response.payload).toContain('processed payment requests')
+  })
+
+  test('GET /monitoring/view-processed-payment-requests route returns 404 if V2 events disabled', async () => {
```
Comment (Kirsten1822) [inline test/integration/narrow/routes/monitoring.test.js:311]: Does it only return 404 or does it also route to another view when V2 events disabled?

File: test/unit/payments/get-payments-by-scheme-id.test.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+jest.mock('../../../app/payments/get-data')
+const { getData: mockGetData } = require('../../../app/payments/get-data')
+
+const { SCHEME_ID: SCHEME_ID_CATEGORY } = require('../../../app/constants/categories')
+const { SCHEME_ID_VALUE } = require('../../../app/constants/scheme-id-value')
+
+const { DATA } = require('../../mocks/values/data')
```
Comment (Kirsten1822) [inline test/unit/payments/get-payments-by-scheme-id.test.js:7]: I'd put the mock above the constants and separate the with a space

File: test/unit/payments/get-payments-by-scheme-id.test.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+const { getPaymentsBySchemeId } = require('../../../app/payments')
+
+describe('get payments by frn', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+    mockGetData.mockResolvedValue(DATA)
```
Comment (Kirsten1822) [inline test/unit/payments/get-payments-by-scheme-id.test.js:14]: Space needed here

File: test/unit/payments/get-payments-by-scheme-id.test.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  })
+
+  test('should get data with FRN category and value', async () => {
```
Comment (Kirsten1822) [inline test/unit/payments/get-payments-by-scheme-id.test.js:17]: Payment data? What data are we talking about here?

File: test/unit/payments/get-payments-by-scheme-id.test.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    await getPaymentsBySchemeId()
+    expect(mockGetData).toHaveBeenCalledWith(SCHEME_ID_CATEGORY, SCHEME_ID_VALUE)
+  })
+
+  test('should return data', async () => {
```
Comment (Kirsten1822) [inline test/unit/payments/get-payments-by-scheme-id.test.js:22]: See my comment here: https://github.com/DEFRA/ffc-pay-web/pull/50/files#r1212859981

File: app/payments/get-payments-by-scheme-id.js (hunk 1 above)
Comment (johnwatson484) [reply app/payments/get-payments-by-scheme-id.js:5]: The Data Hub requires both a `category` and a `value` for every request it gets, otherwise it rejects the message.

So the options are:
1. Update the Data Hub's validation and just pass the `value` as `undefined`.
2. Pass a default value to pass the validation (what you are doing).

I feel that 2 is probably the corect approach, but I think the value being set to '1' is misleading as it wrongly implies other numbers can be passed to get different results.  

I'd suggest changing the value to `all` or `default` or something that indicates every scheme will be returned.

Also not sure if we should refer to the function, variable and constant as `schemeId` as we're not actually getting or filtering data based on schemeId.  Would `getPaymentsByScheme`, `SCHEME_VALUE`, `scheme` be better?

File: test/integration/narrow/routes/monitoring.test.js (hunk 4 above)
Comment (simonnineteen90) [reply test/integration/narrow/routes/monitoring.test.js:311]: i think threre is 404 page that it returns, i followed the format of the tests above for these ones.

File: test/unit/payments/get-payments-by-scheme-id.test.js (hunk 6 above)
Comment (simonnineteen90) [reply test/unit/payments/get-payments-by-scheme-id.test.js:14]: where sorry? im getting a linting error when putting a space between the `describe` and `beforeEach`

File: test/unit/payments/get-payments-by-scheme-id.test.js (hunk 7 above)
Comment (simonnineteen90) [reply test/unit/payments/get-payments-by-scheme-id.test.js:17]: its mock data for the test but have updated the test name to be more clear.

File: app/payments/get-payments-by-scheme-id.js (hunk 1 above)
Comment (simonnineteen90) [reply app/payments/get-payments-by-scheme-id.js]: gone with option 2 and renamed references

File: app/payments/get-payments-by-scheme.js
Comment (Kirsten1822) [inline app/payments/get-payments-by-scheme.js:1]: Why was this file changed to get-payments-by-scheme.js? If that change is intentional, then the test file associated with this file should also be changed?

File: test/unit/payments/get-payments-by-scheme-id.test.js (hunk 6 above)
Comment (Kirsten1822) [reply test/unit/payments/get-payments-by-scheme-id.test.js:14]: Between jest.clearAllMocks and mockGetData.mockResolvedValue

File: test/unit/payments/get-payments-by-scheme-id.test.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    jest.clearAllMocks()
+    mockGetData.mockResolvedValue(DATA)
+  })
+
+  test('should call mockGetData with schemeId category and value', async () => {
+    await getPaymentsByScheme()
+    expect(mockGetData).toHaveBeenCalledWith(SCHEME_ID_CATEGORY, SCHEME_VALUE)
+  })
+
+  test('should return DATA when getPaymentsBySchemeId is called', async () => {
```
Comment (Kirsten1822) [inline test/unit/payments/get-payments-by-scheme-id.test.js:22]: Why is data capitalised here? Is It supposed to be referencing the same data as the test name above?

File: app/payments/get-payments-by-scheme.js
Comment (simonnineteen90) [reply app/payments/get-payments-by-scheme.js:1]: Johns comment above suggests that we should refer to `scheme` ratehr than `scheme-id` so updated file name. I missed the test file but have renamed now.

File: test/unit/payments/get-payments-by-scheme-id.test.js (hunk 9 above)
Comment (Kirsten1822) [reply test/unit/payments/get-payments-by-scheme-id.test.js]: I guess it's just not very clear what data is or should be

**Review by Kirsten1822 — APPROVED** [review]
Happy to approve :)
